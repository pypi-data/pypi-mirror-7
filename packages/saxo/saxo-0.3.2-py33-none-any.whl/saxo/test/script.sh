#!/bin/bash

# http://inamidst.com/saxo/
# Created by Sean B. Palmer

USER=sax

echo '$' HOME=/tmp/sax
HOME=/tmp/sax
echo

DUX=$(python3 -c 'import os; print(os.path.expanduser("~"))')

if [ $DUX != /tmp/sax ]
then echo Error: '$HOME' is not /tmp/sax
    exit 1
fi

if [ ! -f saxo ]
then echo Error: saxo script not found
    exit 1
fi

function saxo() {
    echo '$' saxo $@
    ./saxo $@
    echo
}

function record() {
    echo '$' $@
    $@
    echo
}


####################################

record : Testing create, 01

record rm -rf /tmp/sax
record touch /tmp/sax

saxo create

record rm /tmp/sax


####################################

record : Testing create, 02

record rm -rf /tmp/sax

record mkdir /tmp/sax

record touch /tmp/sax/.saxo

saxo create


####################################

record : Testing create, 03

record rm -rf /tmp/sax

record mkdir -p /tmp/sax/.saxo

saxo create


####################################

record : Testing create, 04

record rm -rf /tmp/sax

saxo create

saxo create


####################################

record : Testing start, 01

record rm -rf /tmp/sax

mkdir -p /tmp/sax/.saxo

echo garbage > /tmp/sax/.saxo/config

saxo start

saxo stop


####################################

record : Testing start, 02

record rm -rf /tmp/sax

record mkdir -p /tmp/sax/.saxo

record touch /tmp/sax/.saxo/config

record chmod 000 /tmp/sax/.saxo/config

saxo start

saxo stop

# record chmod 644 /tmp/sax/.saxo/config


####################################

record : Testing start, 03

record rm -rf /tmp/sax

record mkdir -p /tmp/sax/.saxo

record touch /tmp/sax/.saxo/config

record chmod 000 /tmp/sax/.saxo

saxo start

record chmod 755 /tmp/sax/.saxo


####################################

record : Testing options, 01

record rm -rf /tmp/sax

mkdir /tmp/sax

saxo

saxo --help

saxo --version | \
    sed -E 's/[0-9]+\.[0-9]+\.[0-9]+-[0-9]+/VERSION/'

saxo create

saxo -f start | head -n 7

echo


####################################

record : Cleanup

record rm -rf /tmp/sax
