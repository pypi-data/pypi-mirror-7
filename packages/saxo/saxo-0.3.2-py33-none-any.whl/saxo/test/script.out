$ HOME=/tmp/sax

$ : Testing create, 01

$ rm -rf /tmp/sax

$ touch /tmp/sax

$ saxo create
saxo: error: could not create the directory `/tmp/sax/.saxo`

The attempt to make this directory recursively, creating any necessary parent
directories in the process, failed. Check permissions on all parent directories
to make sure that we can write here. Also make sure that the parent path is a
directory.

This is the error message that python gave:

    NotADirectoryError
        [Errno 20] Not a directory: '/tmp/sax/.saxo'

$ rm /tmp/sax

$ : Testing create, 02

$ rm -rf /tmp/sax

$ mkdir /tmp/sax

$ touch /tmp/sax/.saxo

$ saxo create
saxo: error: could not create the directory `/tmp/sax/.saxo`

The attempt to make this directory recursively, creating any necessary parent
directories in the process, failed. Check permissions on all parent directories
to make sure that we can write here. Also make sure that the parent path is a
directory.

This is the error message that python gave:

    FileExistsError
        [Errno 17] File exists: '/tmp/sax/.saxo'

$ : Testing create, 03

$ rm -rf /tmp/sax

$ mkdir -p /tmp/sax/.saxo

$ saxo create
saxo: error: the directory `/tmp/sax/.saxo` already exists

You tried to make a directory that already exists. This is usually caused by
saxo already existing through a previous creation command. It may also be
caused by a path conflict, for example if you thought that you needed to
manually create the saxo directory beforehand.

This script won't write new material into any existing directories in case of
data corruption.

$ : Testing create, 04

$ rm -rf /tmp/sax

$ saxo create
Created /tmp/sax/.saxo/config
Modify this file with your own settings, and then run:

    ./saxo start /tmp/sax/.saxo

$ saxo create
saxo: error: the directory `/tmp/sax/.saxo` already exists

You tried to make a directory that already exists. This is usually caused by
saxo already existing through a previous creation command. It may also be
caused by a path conflict, for example if you thought that you needed to
manually create the saxo directory beforehand.

This script won't write new material into any existing directories in case of
data corruption.

$ : Testing start, 01

$ rm -rf /tmp/sax

$ saxo start
Running saxo as PID 41836
This PID is also saved in /tmp/sax/.saxo/pid

$ saxo stop
saxo: error: There is no bot currently running

$ : Testing start, 02

$ rm -rf /tmp/sax

$ mkdir -p /tmp/sax/.saxo

$ touch /tmp/sax/.saxo/config

$ chmod 000 /tmp/sax/.saxo/config

$ saxo start
Running saxo as PID 41844
This PID is also saved in /tmp/sax/.saxo/pid

$ saxo stop
saxo: error: There is no bot currently running

$ : Testing start, 03

$ rm -rf /tmp/sax

$ mkdir -p /tmp/sax/.saxo

$ touch /tmp/sax/.saxo/config

$ chmod 000 /tmp/sax/.saxo

$ saxo start
Error: Unable to connect to internal socket
Check permissions on the config dir

$ chmod 755 /tmp/sax/.saxo

$ : Testing options, 01

$ rm -rf /tmp/sax

$ saxo
saxo 0.14.0

Usage:
    saxo -v
    saxo create [ directory ]
    saxo [ -f ] [ -o filename ] start [ directory ]
    saxo stop [ directory ]
    saxo active [ directory ]

Try `saxo -h` for more detailed usage

$ saxo --help
saxo 0.14.0

Usage:

    saxo -v / --version
        Show the current saxo version

    saxo create [ directory ]
        Create a default configuration file

    saxo [ Options ] start [ directory ]
        Starts a bot. Options:

            -f / --foreground
                Don't run the bot as a daemon

            -o / --output filename
                Redirect stdout and stderr to filename

    saxo stop [ directory ]
        Stops a bot

    saxo active [ directory ]
        Shows whether a bot is active

$ saxo --version
saxo 0.14.0

$ saxo create
Created /tmp/sax/.saxo/config
Modify this file with your own settings, and then run:

    ./saxo start /tmp/sax/.saxo

$ saxo -f start
Connecting to irc.freenode.net:6667
handle: message ('started scheduler',)
IPC message: started scheduler
handle: receiving ()
handle: sending ()
handle: connected ()
Exception BrokenPipeError: BrokenPipeError(32, 'Broken pipe') in <encodings.utf_8.StreamWriter object at 0x106263450> ignored

$ : Cleanup

$ rm -rf /tmp/sax

